src/utils/utils.tsx

```diff
@@ -0,0 +1,39 @@
+import { Todo } from '../types/Todo';
+
+export enum FilterEnum {
+  ALL = 'all',
+  COMPLETE = 'completed',
+  ACTIVE = 'active',
+
+}
+
+export function getFilterTodos(todos:Todo[], filter:string):Todo[] {
+  const preparedTodos = [...todos];
+
+  if (filter === FilterEnum.ALL) {
+    return preparedTodos;
+  }
+
+  if (filter === FilterEnum.COMPLETE) {
+    return preparedTodos.filter(todo => todo.completed);
+  }
+
+  if (filter === FilterEnum.ACTIVE) {
+    return preparedTodos.filter(todo => !todo.completed);
+  }
+
+  return preparedTodos;
+}
+
+export function inputFilterTodos(todos:Todo[], query:string) {
+  const normalizedQuery = query.trim().toLowerCase();
+  let copyTodos = [...todos];
+
+  if (normalizedQuery) {
+    copyTodos = copyTodos.filter(
```

Comment:
there is no need to create a copy (line 30) because `filter` method doesn't mutate the array